gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobTask.java

```diff
@@ -133,7 +134,11 @@ protected void setResultToUserContent(Map<String, String> keyValues) throws IOEx
   public void cancel() {
     log.info("Cancelling planning job {}", this.planningJobId);
     if (launcher != null) {
-      launcher.executeCancellation();
+      try {
+        launcher.cancelJob(launcher.getJobListener());
+      } catch (JobException e) {
+        throw new RuntimeException("Unable to cancel planning job " + this.planningJobId + ": ", e);
```

Comment:
We need to delete the underlying job as well. cancelJob() for now only handles the stop functionality.